Mathlib/Algebra/Category/ModuleCat/OverFilteredColimits.lean

```diff
@@ -0,0 +1,351 @@
+
+lemma rep_smul {c : J} (r : ℛ.obj c) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (colimit.ι ℛ c r) m = smulColimit r m := by
+  delta colimitSMulColimit
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smulColimit.smul_rep]
+  apply hsmul.respect_ι
+  · erw [Concrete.ι_repColimit_eq]
+  · erw [Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+
+protected lemma one_smul (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (1 : colimit (C := ℜ𝔦𝔫𝔤) ℛ) m = m := by
+  let c : J := (inferInstance : IsFiltered J).2.some
+  rw [show (1 : colimit (C := ℜ𝔦𝔫𝔤) ℛ) = colimit.ι ℛ c 1 by
+    rw [map_one], rep_smul, smulColimit.one_smul]
+
+protected lemma mul_smul
+    (r₁ r₂ : colimit (C := ℜ𝔦𝔫𝔤) ℛ) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (r₁ * r₂) m = colimitSMulColimit r₁ (colimitSMulColimit r₂ m) := by
+  classical
+  let O : Finset J :=
+    {  Concrete.indexRepColimit ℛ r₁, Concrete.indexRepColimit ℛ r₂ }
+  let H : Finset ((X : J) ×' (Y : J) ×' (_ : X ∈ O) ×' (_ : Y ∈ O) ×' (X ⟶ Y)) := {}
+  let j : J := IsFiltered.sup O H
+  have eq₁ : r₁ = colimit.ι ℛ j
+      (ℛ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℛ r₁)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₂ : r₂ = colimit.ι ℛ j
+      (ℛ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℛ r₂)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₃ : r₁ * r₂ = colimit.ι ℛ j
+      (ℛ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℛ r₁) *
+       ℛ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℛ r₂)) := by
+    rw [map_mul]
+    erw [colimit.w_apply, colimit.w_apply, Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+  rw [eq₃]
+  conv_rhs => rw [eq₁]; rhs; rw [eq₂]
+  rw [rep_smul, rep_smul, rep_smul, smulColimit.mul_smul]
+
+lemma smul_zero (r : colimit (C := ℜ𝔦𝔫𝔤) ℛ) : colimitSMulColimit (ℳ := ℳ) r 0 = 0 := by
+  rw [show r = colimit.ι ℛ (Concrete.indexRepColimit ℛ r) _ by
+    erw [Concrete.ι_repColimit_eq], rep_smul, smulColimit.smul_zero]
+
+lemma smul_add (r : colimit (C := ℜ𝔦𝔫𝔤) ℛ) (m₁ m₂ : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit r (m₁ + m₂) = colimitSMulColimit r m₁ + colimitSMulColimit r m₂ := by
+  rw [show r = colimit.ι ℛ (Concrete.indexRepColimit ℛ r) _ by
+    erw [Concrete.ι_repColimit_eq], rep_smul, rep_smul, rep_smul, smulColimit.smul_add]
+
+lemma add_smul (r₁ r₂ : colimit (C := ℜ𝔦𝔫𝔤) ℛ) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (r₁ + r₂) m = colimitSMulColimit r₁ m + colimitSMulColimit r₂ m := by
+  classical
+  let O : Finset J :=
+    {  Concrete.indexRepColimit ℛ r₁, Concrete.indexRepColimit ℛ r₂ }
+  let H : Finset ((X : J) ×' (Y : J) ×' (_ : X ∈ O) ×' (_ : Y ∈ O) ×' (X ⟶ Y)) := {}
+  let j : J := IsFiltered.sup O H
+  have eq₁ : r₁ = colimit.ι ℛ j
+      (ℛ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℛ r₁)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
```

Comment:
```suggestion
    rw [colimit.w_apply, Concrete.ι_repColimit_eq]
```
erw is a code smell. Surely you should be avoiding it if you can? Here and goodness knows how many other places.

Reply:
All `erw` has been removed now! I might be hallucinating but I swear not all `erw` can be changed `rw` a few days ago 😕